tests/unit/util/tuple.cpp

```diff
@@ -371,6 +371,21 @@ void tuple_cat_test()
         HPX_TEST_EQ((*hpx::util::get<1>(result)), 1);
         HPX_TEST_EQ((*hpx::util::get<2>(result)), 2);
     }
+
+    // Don't move references unconditionally
```

Comment:
The test is a little bit unclear, however it checks the intended behaviour through the result data type `hpx::util::tuple<int&, int&&>`.